alacritty/src/renderer/mod.rs

```diff
@@ -105,12 +118,17 @@ impl Renderer {
         }
 
         let (shader_version, gl_version, renderer) = unsafe {
-            let renderer = CStr::from_ptr(gl::GetString(gl::RENDERER) as *mut _).to_string_lossy();
-            let shader_version =
-                CStr::from_ptr(gl::GetString(gl::SHADING_LANGUAGE_VERSION) as *mut _)
-                    .to_string_lossy();
-            let gl_version = CStr::from_ptr(gl::GetString(gl::VERSION) as *mut _).to_string_lossy();
-            (shader_version, gl_version, renderer)
+            let shader_version = gl::GetString(gl::SHADING_LANGUAGE_VERSION);
+            if gl::GetError() != gl::NO_ERROR {
+                return Err("OpenGL shader support is missing, cannot continue".into());
+            }
```

Comment:
My point was to sanity check all calls to `gl::GetString` and get an actual error message, because I think gl allows you to format error, like we do in other cases, based on enum variant.

Reply:
I could not find a way to format the error using gl, but I was able to wrap all calls to check for errors and report then with a caller-provided description (which is user-friendlier than a macro would be).